go/libraries/doltcore/doltdb/commit.go

```diff
@@ -25,111 +24,49 @@ import (
 	"github.com/dolthub/dolt/go/store/types"
 )
 
-const (
-	metaField        = datas.CommitMetaField
-	parentsField     = datas.ParentsField
-	parentsListField = datas.ParentsListField
-	rootValueField   = datas.ValueField
-)
-
 var errCommitHasNoMeta = errors.New("commit has no metadata")
 var errHasNoRootValue = errors.New("no root value")
 
 // Commit contains information on a commit that was written to noms
 type Commit struct {
-	vrw      types.ValueReadWriter
-	commitSt types.Struct
-	parents  []types.Ref
+	vrw     types.ValueReadWriter
+	meta    *datas.CommitMeta
+	parents []types.Ref
+	stref   types.Ref
```

Comment:
it seems a little weird for a commit to store a ref to itself, although it is a succinct way to check equality, etc